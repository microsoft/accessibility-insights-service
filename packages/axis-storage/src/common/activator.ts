import { injectable } from 'inversify';

// tslint:disable: no-any no-object-literal-type-assertion prefer-object-spread
export class DynamicObject {
    [key: string]: any;
}

@injectable()
export class Activator {
    public convert<T>(source: any, typeT?: new () => T): T {
        if (typeT === undefined) {
            const target = this.createInstance(DynamicObject);

            return <T>Object.assign(target, source);
        } else {
            const target = this.createInstance(typeT);

            return <T>Object.assign(target, source);
        }
    }

    public createInstance<T>(typeT: new () => T): T {
        return new typeT();
    }
}
