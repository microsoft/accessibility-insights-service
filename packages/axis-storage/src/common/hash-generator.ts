import { injectable } from 'inversify';
import * as sha256 from 'sha.js';

@injectable()
export class HashGenerator {
    public constructor(private readonly sha: typeof sha256 = sha256) {}

    public generateBase64Hash(...values: string[]): string {
        const hashSeed: string = values.join('|').toLowerCase();

        return this.sha('sha256')
            .update(hashSeed)
            .digest('hex');
    }

    public getPageScanResultDocumentId(baseUrl: string, url: string, runTimeValue: number): string {
        // preserve parameters order for the hash compatibility
        return this.generateBase64Hash(baseUrl, url, runTimeValue.toString());
    }

    public getScanResultDocumentId(scanUrl: string, selector: string, html: string, resultId: string): string {
        // preserve parameters order for the hash compatibility
        return this.generateBase64Hash(scanUrl, selector, html, resultId);
    }

    public getWebsiteDocumentId(baseUrl: string): string {
        // preserve parameters order for the hash compatibility
        return this.generateBase64Hash(baseUrl);
    }

    public getWebsitePageDocumentId(baseUrl: string, url: string): string {
        // preserve parameters order for the hash compatibility
        return this.generateBase64Hash(baseUrl, url);
    }
}
