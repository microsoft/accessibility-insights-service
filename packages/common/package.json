{
    "name": "common",
    "version": "1.0.0",
    "description": "This project welcomes contributions and suggestions.  Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://cla.microsoft.com.",
    "scripts": {
        "build": "tsc && echo",
        "cbuild": "npm-run-all --serial clean build",
        "clean": "rimraf dist test-results",
        "lint": "eslint -c ../../.eslintrc.js --ext .ts ./",
        "lint:fix": "eslint --fix -c ../../.eslintrc.js --ext .ts ./",
        "test": "jest --coverage --colors"
    },
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Microsoft/accessibility-insights-service.git"
    },
    "main": "dist/index.js",
    "author": "Microsoft",
    "license": "MIT",
    "bugs": {
        "url": "https://github.com/Microsoft/accessibility-insights-service/issues"
    },
    "homepage": "https://github.com/Microsoft/accessibility-insights-service#readme",
    "devDependencies": {
        "@types/convict": "^5.2.1",
        "@types/jest": "^26.0.14",
        "@types/node": "^12.12.54",
        "@types/sha.js": "^2.4.0",
        "@types/wtfnode": "^0.7.0",
        "jest": "^26.4.2",
        "jest-circus": "^26.4.0",
        "jest-junit": "^12.0.0",
        "rimraf": "^3.0.2",
        "ts-jest": "^26.3.0",
        "typemoq": "^2.1.0",
        "typescript": "^4.0.2"
    },
    "dependencies": {
        "@sindresorhus/fnv1a": "^2.0.1",
        "convict": "^6.0.0",
        "got": "^11.7.0",
        "inversify": "^5.0.1",
        "lodash": "^4.17.20",
        "reflect-metadata": "^0.1.13",
        "serialize-error": "^7.0.1",
        "sha.js": "^2.4.11",
        "uuid-with-v6": "^1.1.2",
        "wtfnode": "^0.8.3"
    }
}
