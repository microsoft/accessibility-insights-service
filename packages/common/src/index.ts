// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export { HashGenerator } from './ciphers/hash-generator';
export { IoC } from './dependency-injection/setup-singleton-provider';
export { System } from './system/system-utils';
export { Activator } from './system/activator';
export {
    QueueRuntimeConfig,
    RuntimeConfig,
    ScanRunTimeConfig,
    TaskRuntimeConfig,
    ServiceConfiguration,
} from './configuration/service-configuration';
