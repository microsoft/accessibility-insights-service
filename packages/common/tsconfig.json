{
    "extends": "../../tsconfig.base.json",
    "compilerOptions": {
        "target": "esnext",
        "outDir": "dist"
    }
}
