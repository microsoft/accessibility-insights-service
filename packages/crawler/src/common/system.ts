// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as utils from 'util';
import { serializeError as serializeErrorExt } from 'serialize-error';

/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types */

export namespace System {
    export async function wait(timeoutMillisecond: number): Promise<void> {
        await new Promise((resolve) => setTimeout(resolve, timeoutMillisecond));
    }

    export function serializeError(error: any): string {
        return utils.inspect(serializeErrorExt(normalizeHttpResponse(error)), false, null);
    }

    /**
     * Normalizes request object from the HTTP response object
     * @param httpResponse The HTTP response object
     */
    export function normalizeHttpResponse(httpResponse: any): any {
        if (httpResponse === undefined || (httpResponse.request === undefined && httpResponse.response === undefined)) {
            return httpResponse;
        }

        const getRequest = (request: any) => {
            const { headers, body, operationSpec, ...requestCopy } = request;

            return requestCopy;
        };

        const getResponse = (response: any) => {
            const { headers, request, ...responseCopy } = response;

            return responseCopy;
        };

        const { request, response, ...httpResponseCopy } = httpResponse;
        if (request) {
            httpResponseCopy.request = getRequest(request);
        }

        if (response) {
            httpResponseCopy.response = getResponse(response);
        }

        return httpResponseCopy;
    }

    /**
     * Returns elapsed time since the given timestamp, in msec
     * @param timestamp The timestamp to use as start timestamp to calculate elapsed time, in msec.
     * Use ```System.getTimestamp()``` to get timestamp value
     */
    export function getElapsedTime(timestamp: number): number {
        return getTimestamp() - timestamp;
    }

    /**
     * Returns current time, in msec
     */
    export function getTimestamp(): number {
        return Number(process.hrtime.bigint() / 1000000n);
    }
}
