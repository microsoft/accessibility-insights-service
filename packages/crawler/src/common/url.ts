// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as normalizeUrlExt from 'normalize-url';

export namespace Url {
    export function normalizeUrl(url: string): string {
        return normalizeUrlExt.default(url, { stripHash: true, removeQueryParameters: false });
    }
}
