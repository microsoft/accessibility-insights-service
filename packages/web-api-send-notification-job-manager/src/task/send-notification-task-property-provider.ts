// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BatchServiceModels } from '@azure/batch';
import { BatchTaskPropertyProvider } from 'azure-services';
import { EnvironmentSettings, ServiceConfiguration } from 'common';
import { inject, injectable } from 'inversify';

@injectable()
export class SendNotificationTaskPropertyProvider extends BatchTaskPropertyProvider {
    constructor(
        @inject(EnvironmentSettings) private readonly environmentSettings: EnvironmentSettings,
        @inject(ServiceConfiguration) serviceConfig: ServiceConfiguration,
    ) {
        super(serviceConfig, 'start-web-api-send-notification-runner.sh', ['scanId', 'scanNotifyUrl', 'runStatus', 'scanStatus']);
    }

    public getEnvironmentSettings(): BatchServiceModels.EnvironmentSetting[] {
        return [
            {
                name: 'APPINSIGHTS_INSTRUMENTATIONKEY',
                value: this.environmentSettings.getValue('APPINSIGHTS_INSTRUMENTATIONKEY'),
            },
            {
                name: 'KEY_VAULT_URL',
                value: this.environmentSettings.getValue('KEY_VAULT_URL'),
            },
            {
                name: 'AZURE_STORAGE_SCAN_QUEUE',
                value: this.environmentSettings.getValue('AZURE_STORAGE_SCAN_QUEUE'),
            },
        ];
    }

    protected getResourceContainerNames(): string[] {
        return [this.environmentSettings.getValue('RUNNER_SCRIPTS_CONTAINER_NAME'), 'runtime-configuration'];
    }
}
