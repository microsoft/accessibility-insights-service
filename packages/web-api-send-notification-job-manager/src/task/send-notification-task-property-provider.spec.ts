// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import 'reflect-metadata';

import { EnvironmentSettings, ServiceConfiguration } from 'common';
import { IMock, Mock } from 'typemoq';
import { SendNotificationTaskPropertyProvider } from './send-notification-task-property-provider';

// tslint:disable: no-object-literal-type-assertion

describe(SendNotificationTaskPropertyProvider, () => {
    let testSubject: SendNotificationTaskPropertyProvider;
    let environmentSettingsMock: IMock<EnvironmentSettings>;
    const appInsightsKey = 'app insights key1';
    const keyVaultUrl = 'key vault url1';
    const runnerScriptContainerName = 'runner script container';
    let serviceConfigMock: IMock<ServiceConfiguration>;

    beforeEach(() => {
        serviceConfigMock = Mock.ofType(ServiceConfiguration);

        environmentSettingsMock = Mock.ofType(EnvironmentSettings);
        environmentSettingsMock.setup((s) => s.getValue('APPINSIGHTS_INSTRUMENTATIONKEY')).returns(() => appInsightsKey);
        environmentSettingsMock.setup((s) => s.getValue('KEY_VAULT_URL')).returns(() => keyVaultUrl);
        environmentSettingsMock.setup((s) => s.getValue('RUNNER_SCRIPTS_CONTAINER_NAME')).returns(() => runnerScriptContainerName);
        environmentSettingsMock.setup((s) => s.getValue('AZURE_STORAGE_NOTIFICATION_QUEUE')).returns(() => 'notification-queue-name');

        testSubject = new SendNotificationTaskPropertyProvider(environmentSettingsMock.object, serviceConfigMock.object);
    });

    it('builds command line', () => {
        const data = { scanId: 'id1', scanNotifyUrl: 'url1', runStatus: 'completed', scanStatus: 'failed', somOtherArg: 'some value' };

        expect(testSubject.getCommandLine(JSON.stringify(data))).toMatchSnapshot();
    });

    it('returns resourceFiles', () => {
        expect(testSubject.getResourceFiles()).toMatchSnapshot();
    });

    describe('environment settings', () => {
        it('builds environment settings', () => {
            expect(testSubject.getEnvironmentSettings()).toMatchSnapshot();
        });
    });
});
