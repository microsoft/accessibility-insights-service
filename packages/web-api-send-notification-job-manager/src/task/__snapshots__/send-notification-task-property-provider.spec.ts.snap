// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendNotificationTaskPropertyProvider builds command line 1`] = `"/bin/bash -c 'start-web-api-send-notification-runner.sh \\"id1\\" \\"url1\\" \\"completed\\" \\"failed\\"'"`;

exports[`SendNotificationTaskPropertyProvider environment settings builds environment settings 1`] = `
Array [
  Object {
    "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
    "value": "app insights key1",
  },
  Object {
    "name": "KEY_VAULT_URL",
    "value": "key vault url1",
  },
  Object {
    "name": "AZURE_STORAGE_SCAN_QUEUE",
    "value": "notification-queue-name",
  },
]
`;

exports[`SendNotificationTaskPropertyProvider returns resourceFiles 1`] = `
Array [
  Object {
    "autoStorageContainerName": "runner script container",
  },
  Object {
    "autoStorageContainerName": "runtime-configuration",
  },
]
`;
